inspector/contracts/InspectorConsumer.sol

```diff
@@ -0,0 +1,127 @@
+import {IVRFCoordinator} from "@bisonai/orakl-contracts/src/v0.1/interfaces/IVRFCoordinator.sol";
+
+contract InspectorConsumer is InspectorConsumerBase{
+    using Orakl for Orakl.Request;
+
+    uint256 public sRandomWord;
+    uint128 public sResponse;
+    address private sOwner;
+
+    IVRFCoordinator vrfCoordinator;
+    IAggregatorRouter internal router;
+
+    error OnlyOwner(address notOwner);
+
+    modifier onlyOwner() {
+        if (msg.sender != sOwner) {
+            revert OnlyOwner(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address aggregatorRouter, address _rrCoordinator, address _vrfCoordinator)InspectorConsumerBase(_vrfCoordinator, _rrCoordinator){
+        sOwner = msg.sender;
+        router = IAggregatorRouter(aggregatorRouter);
+        vrfCoordinator = IVRFCoordinator(_vrfCoordinator);
+    }
+
+    // Receive remaining payment from requestDataPayment
+    receive() external payable {}
+
+    function requestDataFeed(string calldata pair) public view returns (uint80 roundId, int256 answer){
+        (
+            uint80 roundId_,
+            int256 answer_
+            , /* uint startedAt */
+            , /* uint updatedAt */
+            , /* uint80 answeredInRound */
+        ) = router.latestRoundData(pair);
+        return (roundId_, answer_);
+    }
+
+    function decimals(string calldata pair) public view returns (uint8) {
+        return router.decimals(pair);
+    }
+
+    function requestRR(
+        uint64 accId,
+        uint32 callbackGasLimit
+    ) public onlyOwner returns (uint256 requestId) {
+        bytes32 jobId = keccak256(abi.encodePacked("uint128"));
+        uint8 numSubmission = 1;
+
+        Orakl.Request memory req = buildRequest(jobId);
+        req.add("get", "https://api.coinbase.com/v2/exchange-rates?currency=BTC");
+        req.add("path", "data,rates,USDT");
+        req.add("pow10", "8");
+
+        requestId = rrCoordinator.requestData(req, callbackGasLimit, accId, numSubmission);
+    }
+
+    function requestRRDirect(
+        uint32 callbackGasLimit
+    ) public payable returns (uint256 requestId) {
+        bytes32 jobId = keccak256(abi.encodePacked("uint128"));
+        uint8 numSubmission = 1;
+
+        Orakl.Request memory req = buildRequest(jobId);
+        req.add("get", "https://api.coinbase.com/v2/exchange-rates?currency=BTC");
+        req.add("path", "data,rates,USDT");
+        req.add("pow10", "8");
+
+        requestId = rrCoordinator.requestData{value: msg.value}(
+            req,
+            callbackGasLimit,
+            numSubmission,
+            address(this)
+        );
+    }
+
+    function requestVRF(
+        bytes32 keyHash,
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint32 numWords
+    ) public onlyOwner returns (uint256 requestId) {
+        requestId = vrfCoordinator.requestRandomWords(keyHash, accId, callbackGasLimit, numWords);
+    }
+
+    function requestVRFDirect(
```

Comment:
This as well, I think if everything works, it would work for both regular and direct payments. We can remove this to save some gas

Reply:
I felt bothered to set account for regular payment and setting up script and environment variables for regular payment. If you insist, I'll update codes for regular payment.

Reply:
There's one more consideration. `onlyOwner` modifier are in regular request functions, that means everyone has to have shared deployer PK to use regular payment

Reply:
It's crucial to test regular account fulfillment because we encourage users to user regular account, not direct payment. And I beleive most of our users use regular account

Reply:
I see, thank you for the feedback, I'll update with regular payment then